manifests/dcgm/telegraf.pp

```diff
@@ -0,0 +1,131 @@
+# @summary Configures telegraf reporting for DCGM
+#
+# @param dcgm_telegraf_port
+#   Port that telegraf socket will listen on, defaults to 8094
+#
+# @param dcgm_telegraf_py_port
+#   Port that DCGM python process will listen on, defaults to 5556
+#
+# @param enable
+#   Enable or disable telegraf reporting for DCGM
+#
+# @example
+#   include profile_gpu::dcgm::telegraf
+class profile_gpu::dcgm::telegraf (
+  Integer $dcgm_telegraf_port,
+  Integer $dcgm_telegraf_py_port,
+  Boolean $enable,
+) {
+
+  if ($enable) {
+    $ensure_parm = 'present'
+  } else {
+    $ensure_parm = 'absent'
+  }
+
+  #
+  # Setup nvidia-dcgm systemd service
+  #
+  systemd::unit_file { 'nvidia-dcgm.service':
+    content => file("${module_name}/nvidia-dcgm.service"),
+    enable  => $enable,
+    active  => $enable,
+    before  => Systemd::Unit_File['dcgmd-telegraf.service'],
+    require => Package['datacenter-gpu-manager'],
+  }
+
+  #
+  # Setup dcgmd-telegraf
+  #
+
+  # Setup variables
+  if find_file('/usr/bin/python3') {
+    $exec_start = '/usr/bin/python3'
+    $dcgm_telegraf_py_path = '/usr/local/dcgm/bindings/python3/dcgm_telegraf.py'
+  } elsif find_file('/usr/bin/python') {
+    $exec_start = '/usr/bin/python'
+    $dcgm_telegraf_py_path = '/usr/local/dcgm/bindings/dcgm_telegraf.py'
+  } else {
+    fail('Unable to determine python version')
+  }
+
+  # TMP fix in place for some bug in NVIDIA DCGM
+  file_line { 'fix_dcgm_telegraf_py':
+    path               => $dcgm_telegraf_py_path,
+    match              => '        self\.m_sock.sendto\(payload, self\.m_dest\)',
+    line               => '        self.m_sock.sendto(payload.encode(), self.m_dest)',
+    append_on_no_match => 'false',
+    require            => Package['datacenter-gpu-manager'],
+    notify             => Service['dcgmd-telegraf.service'],
+  }
+
+  # First Modification so dcgmd-telegraf listens on a static port and only on localhost
+  file_line { 'dcgm_telegraf_py_localhost_listen_only':
+    path               => $dcgm_telegraf_py_path,
+    after              => 'DEFAULT_TELEGRAF_PORT = 8094',
+    # Cannot add them both in one go, line isn't smart enough to check multiple lines
+    # So you instead end up added the two lines every time puppet runs
+    line               => "LISTEN_HOST = '127.0.0.1'",
+    append_on_no_match => 'false',
+    require            => Package['datacenter-gpu-manager'],
+    notify             => Service['dcgmd-telegraf.service'],
+  }
+
+  # Second Modification so dcgmd-telegraf listens on a static port and only on localhost
+  file_line { 'dcgm_telegraf_py_localhost_listen_only_part2':
+    path               => $dcgm_telegraf_py_path,
+    after              => "LISTEN_HOST = '127.0.0.1'",
+    line               => "LISTEN_PORT = ${dcgm_telegraf_py_port}",
```

Comment:
I think we need a `match` parameter here:
```suggestion
    line               => "LISTEN_PORT = ${dcgm_telegraf_py_port}",
    match              => "LISTEN_PORT.*",
```

Reply:
That `file_line` is adding a new line instead of replacing an existing line. So the after `param` takes the place of a `match` param

Reply:
Actually I take that back. In the case of this particular `file_line` it's useful since the `line` takes a variable. Adding `match` lets it update the port number if someone changed the hiera data after puppet already ran once on the node.

Added a `match` line, and removed the `append_on_no_match` since we actually want it to add a line if match returns nothing.